packages/discovery-provider/src/api/v1/users.py

```diff
@@ -3007,3 +3011,160 @@ def get(self, id, authed_user_id):
         muted_users = get_muted_users(decoded_id)
         muted_users = list(map(extend_user, muted_users))
         return success_response(muted_users)
+
+
+encrypted_email_response = make_response(
+    "encrypted_email_response", ns, fields.List(fields.Nested(encrypted_email))
+)
+
+USER_ENCRYPTED_EMAILS_ROUTE = "/<string:id>/encrypted_emails"
+
+
+@ns.route(USER_ENCRYPTED_EMAILS_ROUTE)
+class UserEncryptedEmails(Resource):
+    @record_metrics
+    @ns.doc(
+        id="""Get Encrypted Emails""",
+        description="""Gets the encrypted emails where the given user is either the owner or has primary access""",
+        params={"id": "A User ID"},
+        responses={
+            200: "Success",
+            400: "Bad request",
+            401: "Unauthorized",
+            403: "Forbidden",
+            500: "Server error",
+        },
+    )
+    @ns.marshal_with(encrypted_email_response)
+    def get(self, id):
+        decoded_id = decode_with_abort(id, ns)
+
+        db = get_db_read_replica()
+        with db.scoped_session() as session:
+            emails = (
+                session.query(EncryptedEmail)
+                .filter(
+                    EncryptedEmail.primary_user_id == decoded_id,
+                )
+                .all()
+            )
+
+            emails = [
+                {
+                    "id": email.id,
+                    "email_owner_user_id": email.email_owner_user_id,
+                    "primary_user_id": email.primary_user_id,
+                    "encrypted_email": email.encrypted_email,
+                    "created_at": email.created_at,
+                    "updated_at": email.updated_at,
+                }
+                for email in emails
+            ]
+
+            return success_response(emails)
+
+
+email_encryption_key_response = make_response(
+    "email_encryption_key_response", ns, fields.Nested(email_encryption_key)
+)
+
+USER_EMAIL_ENCRYPTION_KEY_ROUTE = "/<string:id>/email_encryption_key"
```

Comment:
super duper nit but I prefer using the string literals in the `@ns.route` command